conjure-lib/src/main/java/com/palantir/conjure/java/lib/DoubleArrayList.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.lib;
+
+import java.util.AbstractList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.RandomAccess;
+
+public final class DoubleArrayList extends AbstractList<Double> implements RandomAccess {
```

Comment:
@carterkozak would you like this to be internal or made public? Currently the implementation is pretty trimmed down to mostly what is accessible through the generated code.

If ur team is cool owning maintenance and thinks this will be useful in other places, I'm happy to own flushing out the implementation to optimize all the other methods.

Reply:
> would you like this to be internal or made public?

This should be package private and only accessible as a `List<Double>` result from the public factory methods in `ConjureCollections`.

>  Currently the implementation is pretty trimmed down to mostly what is accessible through the generated code.

Do you have documentation describing what it does and doesn't support?

Reply:
This is how the class is accessed from generated code. So minimum required functionality is `add`, `addAll`, `constructor`, and `get`. On top of that remove, clear, and set are implemented, but won't ever really be used. AbstractList gives us `iterator` and `indexOf`

```
@JsonSetter(value = "doubleItems", nulls = Nulls.SKIP, contentNulls = Nulls.FAIL)
public Builder doubleItems(@Nonnull Iterable<Double> doubleItems) {
    checkNotBuilt();
    this.doubleItems = ConjureCollections.newDoubleArrayList(
            Preconditions.checkNotNull(doubleItems, "doubleItems cannot be null"));
    return this;
}

public Builder addAllDoubleItems(@Nonnull Iterable<Double> doubleItems) {
    checkNotBuilt();
    ConjureCollections.addAll(
            this.doubleItems, Preconditions.checkNotNull(doubleItems, "doubleItems cannot be null"));
    return this;
}

public Builder doubleItems(double doubleItems) {
    checkNotBuilt();
    this.doubleItems.add(doubleItems);
    return this;
}
```

Reply:
The codegen'd getter returns List<Double>
```
@JsonProperty("doubleItems")
public List<Double> getDoubleItems() {
    return this.doubleItems;
}
```